src/cpp/Cdr.cpp

```diff
@@ -2754,10 +2879,919 @@ Cdr& Cdr::deserializeWStringSequence(
+                        }
+                        break;
+                    default:
+                        assert(false); // header_serialized_ must have only SHORT_HEADER or LONG_HEADER
+                }
+            }
+
+            jump(member_serialized_size);
+        }
+        else
+        {
+            // Use inner type DHEADER as NEXTINT
+            switch (current_state.header_serialized_)
+            {
+                case XCdrHeaderSelection::SHORT_HEADER:
+                    if (AUTO_WITH_SHORT_HEADER_BY_DEFAULT == current_state.header_selection_)
+                    {
+                        xcdr2_end_long_member_header(current_state.next_member_id_, 0);
+                        offset_ = last_offset;
+                    }
+                    else
+                    {
+                        throw BadParamException("Cannot encode XCDRv2 LongMemberHeader");
+                    }
+                    break;
+                case XCdrHeaderSelection::LONG_HEADER:
+                    xcdr2_shrink_to_long_member_header(current_state.next_member_id_, last_offset);
+                    offset_ = last_offset;
+                    offset_ -= sizeof(uint32_t);
+                    break;
+                default:
+                    assert(false); // header_serialized_ must have only SHORT_HEADER or LONG_HEADER
+            }
+
+            m_lastDataSize = 0;
+        }
+    }
+
+    next_member_id_ = MEMBER_ID_INVALID;
+    serialized_member_size_ = NO_SERIALIZED_MEMBER_SIZE;
+
+    return *this;
+}
+
+void Cdr::xcdr2_deserialize_member_header(
+        MemberId& member_id,
+        Cdr::state& current_state)
+{
+    uint32_t flags_and_member_id = 0;
+    deserialize(flags_and_member_id);
+    member_id.must_understand = (flags_and_member_id & 0x80000000);
+    uint32_t mid = (flags_and_member_id & 0x0FFFFFFF);
+    uint32_t lc = (flags_and_member_id & 0x70000000) >> 28;
+
+    member_id.id = mid;
+
+    if (4 > lc)
+    {
+        switch (lc)
+        {
+            case 0:
+                current_state.member_size_ = 1;
+                break;
+            case 1:
+                current_state.member_size_ = 2;
+                break;
+            case 2:
+                current_state.member_size_ = 4;
+                break;
+            case 3:
+                current_state.member_size_ = 8;
+                break;
+            default:
+                break;
+        }
+        current_state.header_serialized_ = XCdrHeaderSelection::SHORT_HEADER;
+    }
+    else if (4 == lc)
+    {
+        uint32_t size = 0;
+        deserialize(size);
+        current_state.member_size_ = size;
+        current_state.header_serialized_ = XCdrHeaderSelection::LONG_HEADER;
+    }
+    else
+    {
+        uint32_t size = 0;
+        deserialize(size);
+        current_state.member_size_ = 4 + (size * (5 == lc ? 1 : (6 == lc ? 4 : 8)));
+        current_state.header_serialized_ = XCdrHeaderSelection::SHORT_HEADER;
```

Comment:
```suggestion
        current_state.header_serialized_ = XCdrHeaderSelection::LONG_HEADER;
```

Reply:
No, setting SHORT_HEADER avoid to deserialize DHEADER in the inner type.

Reply:
Set a comment.